tests/cpp/test_gpu_view.cpp

```diff
@@ -2312,4 +2312,103 @@ TEST_F(GpuViewTest, ExpandedBroadcast) {
   testValidate(&fusion, {actual_out_tensor}, {in_tensor}, __LINE__, __FILE__);
 }
 
+// See https://github.com/NVIDIA/Fuser/issues/1126
+TEST_F(GpuViewTest, SplitExpandedBroadcast) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* in = makeContigConcreteTensor({4, 5});
+  fusion.addInput(in);
+  TensorView* out = broadcast(in, {false, false, true});
+  out = expand(
+      out,
+      {IrBuilder::create<Val>(4),
+       IrBuilder::create<Val>(5),
+       IrBuilder::create<Val>(6)});
+  out = reshape(out, {4, 5, 6}, {4, 5, 2, 3});
+  fusion.addOutput(out);
```

Comment:
This is what `schedulePointwise` gives us:
```
T5_l[ iblockIdx.x40{( ceilDiv(( ceilDiv(( ceilDiv(( 4 * ( 5 * ( 2 * ( ceilDiv(6, 2) ) ) ) ), 128) ), 1) ), 1) )}, iUS41{1}, iS39{1}, ithreadIdx.x37{128} ] ca_pos( 4 ) produce_pos( 4 )
 root domain : (iS8{4}, iS9{5}, bS11{1 ex 6}rf)
  Outer split: bS11{1 ex 6}rf by factor 2 -> bS12{1 ex 2}rf, bS13{1 ex ( ceilDiv(6, 2) )}rf, start offset: 0, stop offset: 0
 rfactor domain : (iS8{4}, iS9{5}, bS12{1 ex 2}rf, bS13{1 ex ( ceilDiv(6, 2) )}rf)
 contiguity: t t n n
  Merge: bS12{1 ex 2}rf and bS13{1 ex ( ceilDiv(6, 2) )}rf -> bS33{( 1 * 1 ) ex ( 2 * ( ceilDiv(6, 2) ) )}
  Merge: iS9{5} and bS33{( 1 * 1 ) ex ( 2 * ( ceilDiv(6, 2) ) )} -> iS34{( 5 * ( 2 * ( ceilDiv(6, 2) ) ) )}
  Merge: iS8{4} and iS34{( 5 * ( 2 * ( ceilDiv(6, 2) ) ) )} -> iS35{( 4 * ( 5 * ( 2 * ( ceilDiv(6, 2) ) ) ) )}
  Split: iS35{( 4 * ( 5 * ( 2 * ( ceilDiv(6, 2) ) ) ) )} by factor 128 -> iS36{( ceilDiv(( 4 * ( 5 * ( 2 * ( ceilDiv(6, 2) ) ) ) ), 128) )}, ithreadIdx.x37{128}, start offset: 0, stop offset: 0
  Split: iS36{( ceilDiv(( 4 * ( 5 * ( 2 * ( ceilDiv(6, 2) ) ) ) ), 128) )} by factor 1 -> iS38{( ceilDiv(( ceilDiv(( 4 * ( 5 * ( 2 * ( ceilDiv(6, 2) ) ) ) ), 128) ), 1) )}, iS39{1}, start offset: 0, stop offset: 0
  Split: iS38{( ceilDiv(( ceilDiv(( 4 * ( 5 * ( 2 * ( ceilDiv(6, 2) ) ) ) ), 128) ), 1) )} by factor 1 -> iblockIdx.x40{( ceilDiv(( ceilDiv(( ceilDiv(( 4 * ( 5 * ( 2 * ( ceilDiv(6, 2) ) ) ) ), 128) ), 1) ), 1) )}, iUS41{1}, start offset: 0, stop offset: 0
 leaf domain : (iblockIdx.x40{( ceilDiv(( ceilDiv(( ceilDiv(( 4 * ( 5 * ( 2 * ( ceilDiv(6, 2) ) ) ) ), 128) ), 1) ), 1) )}, iUS41{1}, iS39{1}, ithreadIdx.x37{128})
```
That looks OK to me but this is the CUDA kernel that's generated:
```c++
__global__ void nvfuser_none_f0_c0_r0_g0(Tensor<float, 2, 2> T0, Tensor<float, 4, 4> T3) {
  nvfuser_index_t i0;
  i0 = ((nvfuser_index_t)threadIdx.x) + (128LL * ((nvfuser_index_t)blockIdx.x));
  nvfuser_index_t i1;
  i1 = i0 % 30LL;
  if (((i0 < 120LL) && (i1 < 5LL))) {
    float T4[1LL];
    T4[0LL] = 0LL;
    T4[0LL]
       = T0[((5LL * (i0 / 30LL)) + i1)];
    float T1[1LL];
    T1[0LL]
       = T4[0LL];
    float T2[1LL];
    T2[0LL]
       = T1[0LL];
    float T5[1LL];
    T5[0LL]
       = T2[0LL];
    T3[i0]
       = T5[0LL];
  }
}
```
`i0` is the index into a 120-element vector, as if the [4, 5, 6] sized output was unexpanded. It is used to index the output store. The load and store are predicated such that `i0 % 30 < 5`, but that's not enough to guarantee a proper store to an array of size [4, 5]; for example, `i0 = 90` corresponds to `i1 = i0 % 30 == 0` so we will load element `T0[5 * (i0 / 30) + i1] = T0[15]` and write it to `T3[90]`, which is actually out of bounds assuming we have allocated the output properly as size 20. We don't see this prior to this PR since the output is unexpanded and the `i1 < 5` predicate is absent.

I think this might show that expanded dst indexing is not working properly, as I would expect us to use the same index for `T0` and `T3` here.